### PR TITLE
BR-4592: Shadow test failures on PHP 5.6

### PR DESCRIPTION
**Pull request by alex-vlasov**
Second variant - patching stock Makefile to make extension load order correct

### CODE REVIEW

File: tests/override_class.phpt [hunk 1]
```diff
@@ -5,7 +5,7 @@ zip
 --INI--
 shadow.override=ziparchive::open@w0,ziparchive::addfile@0
 --SKIPIF--
-<?php if (!extension_loaded("shadow")) print "skip"; ?>
+<?php if (!extension_loaded("shadow") && !extension_loaded("zip")) print "skip"; ?>
```
Comment (morozov) [inline tests/override_class.phpt]: Shouldn't it be `||`?

File: .travis.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ before_script:
   - ./configure --with-php-config=php-config
   - make
   - sed -i.stock -r -e 's/!mail_qa_team\(\$[[:alnum:]_]+,[[:space:]]*\$[[:alnum:]_]+,[[:space:]]*\$[[:alnum:]_]+\)/true/' run-tests.php
+  - sed -i.stock -e "s/\$ini_settings\['extension'\]\[\] = \$ext_dir \. DIRECTORY_SEPARATOR \. \$req_ext \. '\.' \. PHP_SHLIB_SUFFIX;/array_unshift\(\$ini_settings\['extension'\], \$ext_dir \. DIRECTORY_SEPARATOR \. \$req_ext \. '\.' \. PHP_SHLIB_SUFFIX\);/" run-tests.php
```
Comment (morozov) [inline .travis.yml:15]: Given that the same replacement works for all PHP 5.4…5.6, we may hope it won't break tomorrow. @alex-vlasov if you think this is unreliable, we can only leave the check for `extension_loaded('zip')` and have the problem solved. I don't insist on this exact approach. But this one still looks correct to me.

File: tests/override_class.phpt (hunk 1 above)
Comment (morozov) [reply tests/override_class.phpt]: I see that all tests in this suite check if `extension_loaded('shadow')`, but what's the point? If we can't load the extension being tested and skip all tests, and mark the suite as passed, does it mean we've tested everything we could? I think it makes sense (not as part of this issue) to try removing this check and seeing if we can avoid it.

File: .travis.yml (hunk 2 above)
Comment (alex-vlasov) [reply .travis.yml:15]: `extension_loaded('zip')` doesn't solve the problem. Extension is loaded when test fails on travis, we can solve it only by correcting load order somehow.

**Pull request merged**